src/components/Sidebar/Tabs/DataTab.tsx

```diff
@@ -106,13 +115,17 @@ export default function DataTab({
     }
     const dtype = columnTypes[field];
 
-    // Delete all filters on the old field.
-    let newSpec = deleteSpecFilter(
-      graphSpec,
-      field,
-      columnTypes[field] === 'quantitative' ? 'range' : 'oneOf',
-      { deleteCompound: true }
-    );
+    let newSpec =
+      // check if the filter is being used in another pill
+      hasDuplicateField(graphSpec, field)
+        ? (Object.assign(graphSpec) as GraphSpec)
```

Comment:
What is happening here?

Reply:
Sorry, I was trying to keep the existing filter here. I fixed it. Thanks for catching this.